test/task.cpp

```diff
@@ -0,0 +1,95 @@
+#include <gtest/gtest.h>
+
+#include <functional>
+#include <yaclib/task.hpp>
+
+namespace {
+
+using namespace yaclib;
+
+int counter = 0;
+void AddOne() {
+  counter++;
+}
+
+struct Idle {
+  void DoNothing() {
+    const_method_called = false;
+  }
+
+  void DoNothing() const {
+    const_method_called = true;
+  };
+
+  void DoNothingForceConst() const {
+    const_method_called = true;
+  }
+
+  bool mutable const_method_called{false};
```

Comment:
- [ ] make three different bool member
- [ ] make three functions for check invariant after calls